keras/engine/training.py

```diff
@@ -962,10 +962,21 @@ def _make_predict_function(self):
                                                name='predict_function',
                                                **kwargs)
 
+    def _check_num_samples(self, ins, batch_size=None, steps=None, steps_name='steps'):
+        if steps is not None:
+            num_samples = steps
```

Comment:
Would the best approach then be the creation of a second code path in `fit()` that works more like `fit_generator()`? Perhaps it would be best to create a new function like `fit_tensor()`?